bin/aws-cli-mfa-fish

```diff
@@ -0,0 +1,182 @@
+    argparser.add_argument('-f', '--aws-creds-file', default=str(Path.home())+'/.aws/credentials',
+        help='file path to AWS credentials file (default: ~/.aws/credentials)')
+    argparser.add_argument('-e', '--aws-env-vars', action='store_true',
+        help='export/set AWS access/secret/session env vars instead of updating creds file')
+    argparser.add_argument('-x', '--no-export-profile', action='store_true',
+        help='do not export/set AWS_PROFILE to the one generated by STS')
+    argparser.add_argument('-d', '--lifetime-duration', type=int,
+        help='do not export/set AWS_PROFILE to the one generated by STS')
+
+    return argparser.parse_args()
+
+
+def gen_sts_cmd(cli_args):
+    sts_args = []
+
+    if cli_args.aws_profile:
+        sts_args.append(f'--profile {cli_args.aws_profile}')
+
+    sts_args.append(f'--serial-number {cli_args.profile_arn}')
+    sts_args.append(f'--token-code {cli_args.mfa_token}')
+
+    if cli_args.lifetime_duration:
+        sts_args.append(f'--duration-seconds {cli_args.lifetime_duration}')
+
+    return f\"aws sts get-session-token {' '.join(sts_args)}\"
+
+
+def write_config(creds_file, mfa_profile_section, creds):
+    config = configparser.ConfigParser()
+    config.read(creds_file)
+
+    config[mfa_profile_section] = creds
+
+    with open(creds_file, 'w') as configfile:
+       config.write(configfile)
+
+
+def apply_sts_json(
+        sts_json,
+        use_env_vars,
+        creds_file,
+        mfa_profile_section,
+        export_profile
+    ):
+
+    envvars = {}
+    creds = {
+        'AWS_ACCESS_KEY_ID': sts_json[\"Credentials\"][\"AccessKeyId\"],
+        'AWS_SECRET_ACCESS_KEY': sts_json[\"Credentials\"][\"SecretAccessKey\"],
+        'AWS_SESSION_TOKEN': sts_json[\"Credentials\"][\"SessionToken\"],
+    }
+
+    if use_env_vars:
+        envvars = creds.copy()
+    else:
+        write_config(creds_file, mfa_profile_section, creds)
+
+        if export_profile:
+            envvars['AWS_PROFILE'] = mfa_profile_section
+
+    return envvars
+
+
+def build_response():
+    response = {}
+
+    cli_args = parse_cli_args()
+    sts_cmd = gen_sts_cmd(cli_args)
+    response['sts_cmd'] = sts_cmd
+
+    sts_output = os.popen(sts_cmd).read()
+    try:
+        sts_json = json.loads(sts_output)
+
+        try:
+            response['envvars'] = apply_sts_json(
+                sts_json,
+                cli_args.aws_env_vars,
+                cli_args.aws_creds_file,
+                cli_args.aws_creds_mfa_section,
+                not cli_args.no_export_profile
+            )
+        except Exception as ex:
+            response['output'] = str(ex)
+    except:
+        response['output'] = sts_output
+
+    return response
+
+if __name__ == '__main__':
+    print(json.dumps(build_response()))
+"
+
+set -l RESPONSE (/usr/bin/env python3 -c "$PYCODE" $argv)
+
+if string match -r "^usage" "$RESPONSE" &> /dev/null
+    echo "$RESPONSE"
+    exit 0
```

Comment:
Same as above with these exits: this is different in the other shells, on purpose, to only quit if sourced

Reply:
As above, not a thing in fish -- resolved